File: src/main/java/com/ultreon/bubbles/render/gui/QLabel.java
package com.ultreon.bubbles.render.gui;

import com.ultreon.hydro.render.Renderer;

import java.awt.*;

public class QLabel extends QComponent {
    private String text;
    private boolean wrapped;

    protected Color foregroundColor;

    public QLabel(String text) {
        this.text = text;

        backgroundColor = new Color(0, 0, 0, 0);
        foregroundColor = Color.white;
    }

    @Override
    public void render(Renderer ngg) {
        Renderer ngg1 = ngg.create(getX(), getY(), getWidth(), getHeight());
        renderComponent(ngg1);
        ngg1.dispose();
    }

    @Override
    public void renderComponent(Renderer ngg) {
        ngg.color(backgroundColor);
        ngg.fill(getBounds());

        ngg.color(foregroundColor);

        if (wrapped) ngg.wrappedText(text, 0, 0, getWidth());
        else ngg.multiLineText(text, 0, 0);
    }

    @Override
    public void tick() {

    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public boolean isWrapped() {
        return wrapped;
    }

    public void setWrapped(boolean wrapped) {
        this.wrapped = wrapped;
    }

    public Color getForegroundColor() {
        return foregroundColor;
    }

    public void setForegroundColor(Color foregroundColor) {
        this.foregroundColor = foregroundColor;
    }
}

File: src/main/kotlin/qtech/bubbles/gui/QLabel.kt
package qtech.bubbles.gui

import qtech.hydro.GraphicsProcessor
import java.awt.Color

class QLabel(var text: String) : QComponent() {
    var isWrapped = false
    var foregroundColor: Color
    override fun render(ngg: GraphicsProcessor) {
        val ngg1 = ngg.create(x, y, width, height)
        renderComponent(ngg1)
        ngg1.dispose()
    }

    override fun renderComponent(ngg: GraphicsProcessor) {
        ngg.color(backgroundColor)
        ngg.shapeF(bounds)
        ngg.color(foregroundColor)
        if (isWrapped) ngg.wrappedText(text, 0, 0, width) else ngg.multilineText(text, 0, 0)
    }

    override fun tick() {}

    init {
        backgroundColor = Color(0, 0, 0, 0)
        foregroundColor = Color.white
    }
}